bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -396,19 +396,21 @@ private static ColorTemperatureCharacteristic createColorTemperatureCharacterist
 
     private static CurrentFanStateCharacteristic createCurrentFanStateCharacteristic(final GenericItem item,
             HomekitAccessoryUpdater updater) {
+        final DecimalType value = item.getStateAs(DecimalType.class);
         return new CurrentFanStateCharacteristic(
-                () -> CompletableFuture.completedFuture(item.getStateAs(DecimalType.class) != null
-                        ? CurrentFanStateEnum.fromCode(item.getStateAs(DecimalType.class).intValue())
+                () -> CompletableFuture.completedFuture( value != null
+                        ? CurrentFanStateEnum.fromCode(value.intValue())
```

Comment:
same question here. will the Supplier call "item.getStateAs" ?

Reply:
good point. I'll change that. There is another problem: if the state has an invalid integer value (e.g. `6`), `fromCode` will return `null`. This is unnoticed because HAP-Java has no null-annotations. 

There are several possible fixes to that: 

(1) we could check if the returned value is null (easiest)
(2) `fromCode` could be adjusted to return a default value when an invalid code is supplied.

I would suggest to do (1) for now.

Reply:
agree. we should go for (1) now and consider (2) for hap fix. however, im not sure whether the default value is a good solution - there will no way for the client, e.g. OH, to get notice when it is out of range for the enum. maybe we should return something like "RESERVED" or "UNDEF" back.